### DISCUSSION TITLE
Should autocomplete (API) really require "valuePath" configuration?

### DISCUSSION THREAD

**Discussion by luanmm** [directus/directus#7016] [Feature Requests]
I'm facing an use case where the API returns something like this:

```json
{
  "results": [
    "option 1",
    "option 2",
    "option 3"
  ]
}
```

But I can't configure the Autocomplete API field interface because it requires a "valuePath".

As I digged into the source code of Directus, the "valuePath" is used as argument to lodash's "get" function, which accepts a blank string (so it returns the own value for that path, which is what I'm expecting to get to list the options here, right?).

So shoud it really be a required field? Or maybe changing it to optional would benefit more use cases as I'm thinking it could? In that case, perhaps it should only be enforced as a blank string ('') instead of null/undefined values (if it is not the default value that comes into the interface component in Vue).

Thanks in advance for joining this discussion! :smiley:

**Comment by rijkvanzanten** [comment]
Heya! Wouldn't the valuePath be `results` in this case? 🤔

**Reply by luanmm** [reply]
Hello, @rijkvanzanten!

I think, in this case, the "resultsPath" is "results" and "valuePath" should be "the own value", or I'm missing something?

My understading is that "resultsPath" should map the results list (array) and, then, "valuePath" should map the item value/text, is it right?

If I'm right, maybe both could be empty (to comprehend even cases where the result is already an array os items like `["option 1", "option 2", ...]`).

Thanks for the fastest answer I've already got in a Github discussion :smile:

**Reply by rijkvanzanten** [reply]
Ohh right right! You're right, I was confusing resultsPath with valuePath 👍🏻 

So to answer your original question:

> So shoud it really be a required field?

The answer is, probably not!

**Reply by luanmm** [reply]
Should I submit a PR after ensuring these scenarios would really work as expected?

**Reply by rijkvanzanten** [reply]
That'd be great! Thanks 🙂 ❤️

**Reply by luanmm** [reply]
Great, @rijkvanzanten :wink:

**Reply by luanmm** [reply]
As PR #7025 now is merged, I think we are all set in this thread, so feel free to close it if you wish! Thanks for the opinion, @rijkvanzanten, and to all the efforts from the Directus team!

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
